src/main/java/wolox/training/models/User.java

```diff
@@ -88,10 +91,11 @@ public void setBooks(List<Book> books) {
 	}
 
 	public List<Book> addBook(Book bookCurrent) {
-		if (books.stream().filter(book -> book.getIsbn().equals(bookCurrent.getIsbn())).findFirst().isPresent()) {
+		if (books.contains(bookCurrent)) {
```

Comment:
For this to work you need to override the equals method of book. Remember that contains will check for Object equality. Please read this article: https://www.baeldung.com/java-comparing-objects , the most important part you need here is in `"3. Object#equals Method"`.